heroku_connect/models/base.py

```diff
@@ -0,0 +1,205 @@
+from django.core import checks
+from django.db import models
+
+from . import fields
+from ..conf import settings
+
+__all__ = ('HerokuConnectModel',)
+
+
+class HerokuConnectModelBase(models.base.ModelBase):
+    """
+    Custom model base class to set ``Meta`` attributes.
+
+    Sets :attr:`Meta.managed<django.db.models.Options.db_table>` to ``True``
+    and sets a default value for :attr:`Meta.db_table<django.db.models.Options.db_table>`.
+    """
+
+    def __new__(mcs, name, bases, attrs):
+        super_new = super(HerokuConnectModelBase, mcs).__new__
+        fake_class = type.__new__(mcs, name, bases, attrs)
+        org_meta_attr = getattr(fake_class, 'Meta', None)
+        new_class = super_new(mcs, name, bases, attrs)
+        if org_meta_attr is None or not hasattr(org_meta_attr, 'db_table'):
+            new_class._meta.db_table = '.'.join(map(lambda x: '"%s"' % x, [
+                settings.HEROKU_CONNECT_SCHEMA,
+                new_class.sf_object_name.lower(),
+            ]))
+        new_class._meta.managed = False
+        return new_class
+
+
+class HerokuConnectModel(models.Model, metaclass=HerokuConnectModelBase):
+    """
+    Base model for Heroku Connect enabled ORM models in Django.
+
+    Example::
+
+        from heroku_connect import models as hc_models
+
+
+        class MyCustomObject(hc_models.HerokuConnectModel):
+            sf_object_name = 'My_Custom_Object__c'
+
+            custom_date = hc_models.DateTime(sf_field_name='Custom_Date__c')
+
+    Note:
+
+        Subclasses have :attr:`Meta.managed<django.db.models.Options.managed>` set to ``True``.
+
+        A default value for :attr:`Meta.db_table<django.db.models.Options.db_table>` is set based
+        on :attr:`settings.HEROKU_CONNECT_SCHEMA<.HerokuConnectAppConf.HEROKU_CONNECT_SCHEMA>`
+        and :attr:`.sf_object_name`.
+
+    """
+
+    sf_object_name = ''
+    """Salesforce object API name."""
+    sf_access = 'read_only'
+    """
+    Heroku Connect Object access level.
+
+    Access to Heroku Connect tables can be either read or write.
+    Read will only sync from Salesforce to PostgreSQL where
+    write will sync both ways.
+
+    Default value is ``read_only``.
+
+    Accepted values:
+        * ``read_only``
+            Synchronize only from Salesforce to PostgreSQL.
+        * ``read_write``
+            Synchronize both ways between Salesforce and PostgreSQL.
+
+    """
+    sf_notify_enabled = True
+    sf_polling_seconds = 120
+    sf_max_daily_api_calls = 30000
+
+    sf_id = fields.ID(sf_field_name='ID', db_column='sfid')
+    system_mod_stamp = fields.DateTime(sf_field_name='SystemModstamp', db_index=True)
+    is_deleted = fields.Checkbox(sf_field_name='IsDeleted')
```

Comment:
This field is not on all objects...on the `User` Salesforce object it doesn't exist...

Reply:
Good catch! You are right. I will make it a regular model field.

Reply:
Wait, I just verified, it is in fact a field on Salesforce https://devcenter.heroku.com/articles/heroku-connect-database-tables#system-columns
I also checked in the configuration (in case the docs are outdated). The field exists.

Reply:
@codingjoe no it doesn't exist for users... and yes the docs do not mention it... can you please check again in the schema?

Reply:
and in the heroku connect config eventually...

Reply:
I did check in the heroku connect config. I will check for users too now.

Reply:
Dude you are right. It's only user. I will write the support on how they decide it's there not not... wtf... Good catch!

Reply:
https://help.heroku.com/sharing/5295ce37-d767-4355-aef7-95f3cde95915

Reply:
👍